### DISCUSSION TITLE
Full tutorials on react-spring

### DISCUSSION THREAD

**Discussion by msagerup** [pmndrs/react-spring#1292] [Support]
Hi.

Can someone share with me full tutorials on react spring or any courses ?

**Comment by jaredcat** [comment]
I was looking for the same thing. Haven't really found anything on udemy yet either

**Comment by joshuaellis** [comment]
There's not necessarily a course afaik, but there are a fair few demos at the bottom of some pages like [here](https://react-spring.io/hooks/use-spring#demos). That can be edited in CodeSandbox so you'll see the full code.

**Reply by algoflows** [reply]
are there any examples using typescript?

**Reply by joshuaellis** [reply]
all those demos use typescript 👍🏼

**Reply by gormonn** [reply]
At first glance, it seems that this is not enough.
For example, when a developer implements an atomic design. And he wants each of its atoms, for example a button, to extend the animated.button and accept all of its props, and that's where the problems arise.

However, this seems unobvious, but nothing prevents you from simply nesting components inside animated.div where necessary.

**Reply by gormonn** [reply]
No, though. It looks like Nesting components in animated.div won't work just like that. For example, when the component is absolutely positioned.

**Reply by joshuaellis** [reply]
Hey @gormonn is there something you're specifically stuck on? I find your comments a bit vague and confusing, but I imagine with a bit more detail they'll probably make more sense to me! If it's not related to this topic though, we could open a new discussion!
